app/models/post.rb

```diff
@@ -1,2 +1,21 @@
 class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  after_create :update_posts_counter
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_posts_counter
+    post = Post.all
+    update(posts_counter: post.count)
+  end
```

Comment:
### Optional

- [ ] Consider using` author.increment!(:posts_counter)` instead of fetching the count and then updating it. This method directly increments the counter in the database, reducing the need for an extra query. It's a more efficient and idiomatic way in ActiveRecord.

#### example

```
def update_posts_counter
  author.increment!(:posts_counter)
end
```